lxd/db/cluster/instance_profiles.mapper.go

```diff
@@ -46,12 +46,18 @@ DELETE FROM instances_profiles WHERE instance_id = ?
 
 // GetProfileInstances returns all available Instances for the Profile.
 // generator: instance_profile GetMany
-func GetProfileInstances(ctx context.Context, tx *sql.Tx, profileID int) ([]Instance, error) {
+func GetProfileInstances(ctx context.Context, tx *sql.Tx, profileID int, filters ...InstanceFilter) ([]Instance, error) {
 	var err error
 
 	// Result slice.
 	objects := make([]InstanceProfile, 0)
 
+	for _, filter := range filters {
+		if filter.ID != nil {
+			return nil, fmt.Errorf("Cannot filter on ID of Instance when retrieving from \"instances_profiles\"")
```

Comment:
Why is this?

Reply:
The IDs would all be overwritten just a few lines down. This function works by fetching a list of `instance` IDs in this case, that are associated with the specified `profile` ID. Then it creates a list of `ProfileFilter` with each of those IDs, so we only return instances that are associated to the original profileID. 

If any filters are specified, all of the instances associated to the profileID, which have the additional constraints are returned. I don't think it would make sense to process the ID here anyway, because you would be calling a function to get the instances that use the profile, but could potentially return an instance that doesn't use that profile.

Reply:
feels like we should have a different filter struct for these functions then.

Reply:
I was worried that would be too confusing/complicated.

For one it's more complexity, there would need to be two filter structs for each of these (`ProfileInstanceFilter` for filtering instances and `InstanceProfileFilter` for filtering profiles).

Also we would have to ensure that other than the `ID` field, a struct like `ProfileInstanceFilter` needs to have all the same field options as `InstanceFilter`, otherwise we might not have a query for that combination.

Reply:
in my view the complexity exists irrespective. better to have it flagged at build time rather than at tun time (possibily post release).

you could have a common struct type that excludes id field and then embed it in the instancefilter struct along with an id field.